src/keys.ts

```diff
@@ -153,7 +153,11 @@ export function computeDerivedKeys(
       throw unknownLocaleError;
     } else if (pluralRule instanceof Intl.PluralRules) {
       const pluralRulesOptions = pluralRule.resolvedOptions();
-      if (pluralRulesOptions.locale !== locale) {
+      const lang = locale.split('-')[0];
+      if (
```

Comment:
I'm wondering if we could/should be even more permissive. I think what we actually want to ensure here is that a plural rule was found for our current locale, not so much that the locale is consistent (because that would be a bug in i18next/Intl).

The comment above suggests that a pluralRule is always returned even for non-existing locales. Is there any property in the pluralRule or in the pluralRuleOptions that we could check to ensure i18next/Intl did resolve a locale and didn't fallback to whatever the default is? If so, I think I would prefer that we check that here instead of some convoluted heuristic. WDYT?

Reply:
For case like `const pluralRule = new Intl.PluralRules('en-US');`
The locale from `pluralRule.resolvedOptions()` ideally should be `en-US`(BCP 47 language tag) based on the spec. ([ref](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/PluralRules/resolvedOptions))
But different browser and node return different result:
```
// chrome 119, node 18
locale: "en" 
// firefox 119
locale: "en-US"
```
It is possible that node return locale: "en-US" in the future, so I think we can allow `en` and `en-US` at the moment. 

Also, the output of resolvedOptions looks like
`{ locale: "en", type: "cardinal", minimumIntegerDigits: 1, minimumFractionDigits: 0, maximumFractionDigits: 3, pluralCategories: Array ["one", "other"], roundingIncrement: 1, roundingMode: "halfExpand", roundingPriority: "auto", trailingZeroDisplay: "auto" }`. No other property shows the default/fallback locale for it...

Reply:
gotcha, thank you for the explanation.